node/lib/util/status_util.js

```diff
@@ -387,11 +388,13 @@ exports.getRepoStatus = co.wrap(function *(repo, options) {
             const treeId = head.treeId();
             tree = yield NodeGit.Tree.lookup(repo, treeId);
         }
+        repo.submoduleCacheAll();
         const status = yield DiffUtil.getRepoStatus(repo,
                                                     tree,
                                                     options.paths,
                                                     options.ignoreIndex,
                                                     options.showAllUntracked);
+        repo.submoduleCacheClear();
```

Comment:
I guess the idea is that the cache is only useful during certain single operations?  If so, can we in general have wrappers for those operations which cache/uncache around them?